File: android/src/main/kotlin/com/apparence/camerawesome/sensors/SensorOrientationListener.java
package com.apparence.camerawesome.sensors;

import io.flutter.plugin.common.EventChannel;

public class SensorOrientationListener implements EventChannel.StreamHandler, SensorOrientation {

    EventChannel.EventSink events;

    @Override
    public void onListen(Object arguments, EventChannel.EventSink events) {
        this.events = events;
    }

    @Override
    public void onCancel(Object arguments) {
        this.events.endOfStream();
        this.events = null;
    }

    @Override
    public void notify(int orientation) {
        if (this.events == null) {
            return;
        }
        switch (orientation) {
            case 0:
                events.success("PORTRAIT_UP");
                break;
            case 90:
                events.success("LANDSCAPE_LEFT");
                break;
            case 180:
                events.success("PORTRAIT_DOWN");
                break;
            case 270:
                events.success("LANDSCAPE_RIGHT");
                break;
        }
    }
}

File: android/src/main/kotlin/com/apparence/camerawesome/sensors/SensorOrientationListener.kt
package com.apparence.camerawesome.sensors

import io.flutter.plugin.common.EventChannel
import io.flutter.plugin.common.EventChannel.EventSink

class SensorOrientationListener : EventChannel.StreamHandler, SensorOrientation {
    var events: EventSink? = null
    override fun onListen(arguments: Any, events: EventSink) {
        this.events = events
    }

    override fun onCancel(arguments: Any?) {
        events?.endOfStream()
        events = null
    }

    override fun onOrientationChanged(orientation: Int) {
        if (events == null) {
            return
        }
        when (orientation) {
            0 -> events!!.success("PORTRAIT_UP")
            90 -> events!!.success("LANDSCAPE_LEFT")
            180 -> events!!.success("PORTRAIT_DOWN")
            270 -> events!!.success("LANDSCAPE_RIGHT")
        }
    }
}